storage/repair_queue_test.go

```diff
@@ -1,155 +0,0 @@
-	storePool.mu.Lock()
-	defer storePool.mu.Unlock()
-
-	storePool.stores = make(map[proto.StoreID]*storeDetail)
-	for _, storeID := range aliveStoreIDs {
-		storePool.stores[storeID] = &storeDetail{
-			dead: false,
-			desc: proto.StoreDescriptor{StoreID: storeID},
-		}
-	}
-	for _, storeID := range deadStoreIDs {
-		storePool.stores[storeID] = &storeDetail{
-			dead: true,
-			desc: proto.StoreDescriptor{StoreID: storeID},
-		}
-	}
-}
-
-// TestRepairQueueNeedsRepair verifies that the needsRepair method correctly
-// identifies when a range requires repair.
-func TestRepairQueueNeedsRepair(t *testing.T) {
-	defer leaktest.AfterTest(t)
-
-	stopper := stop.NewStopper()
-	defer stopper.Stop()
-	clock := hlc.NewClock(hlc.UnixNano)
-	rpcContext := rpc.NewContext(&base.Context{}, clock, stopper)
-	g := gossip.New(rpcContext, gossip.TestInterval, gossip.TestBootstrap)
-	storePool := NewStorePool(g, TestTimeUntilStoreDeadOff, stopper)
-	a := makeAllocator(storePool)
-	replicateQ := makeReplicateQueue(g, a, clock)
-	repairQ := makeRepairQueue(storePool, &replicateQ, clock)
-
-	testCases := []struct {
-		replica        *Replica
-		storePoolAlive []proto.StoreID
-		storePoolDead  []proto.StoreID
-		eNeeds         bool
-		ePriority      int
-		eReplicas      []proto.Replica
-	}{
-		// Test 0: three stores, all alive, nothing to do
-		{
-			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
-			storePoolAlive: []proto.StoreID{1, 2, 3},
-			storePoolDead:  []proto.StoreID{},
-		},
-		// Test 1: three stores, one dead store, repair the store
-		{
-			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
-			storePoolAlive: []proto.StoreID{1, 2},
-			storePoolDead:  []proto.StoreID{3},
-			eNeeds:         true,
-			ePriority:      0,
-			eReplicas:      createReplicaSets([]proto.StoreID{3}),
-		},
-		// Test 2: three stores, two dead stores, can't repair due to not having quorum
-		{
-			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
-			storePoolAlive: []proto.StoreID{1},
-			storePoolDead:  []proto.StoreID{2, 3},
-			ePriority:      1,
-			eReplicas:      createReplicaSets([]proto.StoreID{2, 3}),
-		},
-		// Test 3: five store, two dead stores, both should be repaired
-		{
-			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3, 4, 5}),
-			storePoolAlive: []proto.StoreID{1, 2, 3},
-			storePoolDead:  []proto.StoreID{4, 5},
-			eNeeds:         true,
-			ePriority:      0,
-			eReplicas:      createReplicaSets([]proto.StoreID{4, 5}),
-		},
-		// Test 4: five store, one dead stores, repair the store
-		{
-			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3, 4, 5}),
-			storePoolAlive: []proto.StoreID{1, 2, 3, 4},
-			storePoolDead:  []proto.StoreID{5},
-			eNeeds:         true,
-			ePriority:      -1,
-			eReplicas:      createReplicaSets([]proto.StoreID{5}),
-		},
-		// Test 5: three stores, all alive, but only 1 is in the store pool, nothing to do
```

Comment:
Added a few test cases: multiple dead nodes, nodes which aren't in the storePool.